include/boost/process/detail/posix/executor.hpp

```diff
@@ -445,8 +453,10 @@ child executor<Sequence>::invoke(boost::mpl::false_, boost::mpl::false_)
     if (_ec)
     {
         //if an error occurred we need to reap the child process
-        ::waitpid(this->pid, nullptr, WNOHANG);
+        int block_mode = _msg == "execve failed" ? 0 : WNOHANG;
+        ::waitpid(this->pid, nullptr, block_mode);
         boost::fusion::for_each(seq, call_on_error(*this, _ec));
+        set_error(_ec, _msg);
```

Comment:
what is this for? we already reported the error.

Reply:
Indeed. 

However, on top of setting `_ec` and `_msg`, `set_error()` indirectly calls `throw process_error(ec, msg); `
Because I have extracted `set_error()` out of `_read_error() ` and because `throw process_error` is only called in error related functions in executor.hpp I thought it will be most appropriate. 

Would you prefer me changing this `set_error(_ec, _msg);` to `throw process_error(_ec, _msg); ` ?

Reply:
Neither is correct I fear.